Migrated Config to common

Sindhu-Vasireddy opened the pull request:
## Description

Migrated config from `utils` to `common`. Also, removed torch dependence from `utils/config.py` and moved to `utils/utils.py`

## Type of Change

-   [X] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation update

## Issue Number

Closes #512 
Closes #536 

## Code Compatibility

-   [X] I have performed a self-review of my code

### Code Performance and Testing

-   [X] I ran the `uv run train` and (if necessary) `uv run evaluate` on a least one GPU node and it works 
-   [ ] If the new feature introduces modifications at the config level, I have made sure to have notified the other software developers through Mattermost and updated the paths in the `$WEATHER_GENERATOR_PRIVATE` directory

Scheduled Jobs with run_id: o19n20k5 & l2loi1sd Slurm job id: 12007844 & 12007562

clessig commented on src/weathergen/utils/utils.py:
```diff
@@ -0,0 +1,26 @@
+# (C) Copyright 2025 WeatherGenerator contributors.
```
Is this the only file where the license was missing? We should ref the issue for this.

clessig reviewed (commented):
Looks good. Just a general remark: let's please keep PRs as clean and focused as possible. If you find some issue that should be fixed but that is unrelated to the PR topic, then please open a second PR. This simplified reviewing.

clessig commented:
@Sindhu-Vasireddy : could you update the branch so that we can merge it.

Sindhu-Vasireddy commented:
> > @clessig the PR is now updated. Could you please have a look?
> 
> @Sindhu-Vasireddy : many thanks. Did you try running the code? I get
> 
> ```
>   File "WeatherGenerator/src/weathergen/utils/logger.py", line 16, in <module>
>     from weathergen.utils.config import _load_private_conf
> ModuleNotFoundError: No module named 'weathergen.utils.config'
> ```
> 
> and
> 
> ```
>   File "WeatherGenerator/src/weathergen/model/blocks.py", line 19, in <module>
>     from weathergen.utils.config import get_dtype
> ModuleNotFoundError: No module named 'weathergen.utils.config'
> ```

@clessig Thank you for pointing out somehow when I launched the slurm job yesterday using the script from the private weather gen repo, it did not throw any runtime errors. It looked clean in the .err file. I have now checked all the occurences of the imports with the old format and changed them. However, unfortunately since Juwels is down today I cannot run a test train. If you could in the meantime do a quick check now with the latest that would be helpful, thank you.

Sindhu-Vasireddy commented:
@clessig I have checked(Slurm job id: 12396556, run_id: g4qghis5) that it is starting to train. Could you please review it now, thank you.
